src/main/java/com/jcraft/jsch/KeyExchange.java

```diff
@@ -290,34 +325,43 @@ protected byte[] normalize(byte[] secret) {
     return foo;
   }
 
-  protected boolean verify(String alg, byte[] K_S, int index, byte[] sig_of_H) throws Exception {
-    int i, j;
 
-    i = index;
+  protected boolean verifyKeyExchangeServerSignature(String alg, byte[] K_S, int index,
```

Comment:
Please rename this method back to `verify()` as it is not backwards compatible for users that have created their own KeyExchange implementations.